### DISCUSSION TITLE
Crs of roi in Sampler

### DISCUSSION THREAD

**Discussion by Modexus** [torchgeo/torchgeo#601] [General]
Do I understand correctly that the `crs` of the `BoundingBox` passed to a `Sampler` has to be in the `crs` of the `dataset` it samples from?

Is there some function to create the correct `BoundingBox` easier?
I believe the `mint` and `maxt` also have to be in the timestamp format provided by the `disambiguate_timestamp` function.

**Comment by adamjstewart** [comment]
> Do I understand correctly that the `crs` of the `BoundingBox` passed to a `Sampler` has to be in the `crs` of the `dataset` it samples from?

Yes, this is correct.

> Is there some function to create the correct `BoundingBox` easier?
> I believe the `mint` and `maxt` also have to be in the timestamp format provided by the `disambiguate_timestamp` function.

You could use `dataset.bounds.mint` (and `maxt`, `minx`, etc.) to get an idea of the bounds of the entire dataset, then do the math to compute the region you are interested in (for example, splitting in half vertically for a train/val/test split). Alternatively, you can use a GIS program like QGIS or ArcGIS to compute the bounds of the region you are interested by ensuring that the CRS of the view is the same as your dataset.

**Comment by Modexus** [comment]
Thanks, I didn't think of calculating it from the `dataset.bounds` in the dataloader, I think that should work for my use-case.
